etna/media/blocks.py

```diff
@@ -24,6 +24,9 @@ def get_api_representation(self, value, context=None):
         We use expand_db_html to get any rich text fields as useful HTML,
         rather than the raw database representation.
         """
+        chapters = value.chapters.stream_block.get_api_representation(
+            value.chapters, context
+        )
```

Comment:
Instead of using the actual StreamBlock instance and calling its own function on itself, you can get the data this way:
```
        return {
            "id": value.id,
            "file": value.url,
            "full_url": value.full_url,
            "type": value.type,
            "mime": value.mime(),
            "title": value.title,
            "date": value.date,
            "description": expand_db_html(value.description),
            "transcript": expand_db_html(value.transcript),
            "chapters": sorted(
                [
                    {
                        "time": int(chapter.value["time"]),
                        "heading": chapter.value["heading"],
                        "transcript": expand_db_html(
                            chapter.value["transcript"].source
                        ),
                    }
                    for chapter in value.chapters
                ],
                key=lambda x: x["time"],
            ),
            "width": value.width,
            "height": value.height,
            "duration": value.duration,
        }
```

Reply:
Which do you think is better?

Reply:
I think yours is better. Changed.